### DISCUSSION TITLE
How do I resolve CERTIFICATE_VERIFY_FAILED

### DISCUSSION THREAD

**Discussion by uSkizzik** [WeblateOrg/weblate#9916] [Q&A]
Hi! I'm trying to setup Weblate on my VPS using docker compose and I've encountered some issues.
When I start it up, I get the following error:
```
check stderr | CRITICALS:
check stderr | ?: (weblate.E003) Cannot send e-mail ([SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate (_ssl.c:1002)), please check EMAIL_* settings.
check stderr |  HINT: https://docs.weblate.org/en/weblate-4.18.2/admin/install.html#out-mail
```

How would I resolve that? I'm trying to connect to the docker host via 172.17.0.1, although I have also tried connecting to the mail service through the internet (mail.skizzium.com). Both return the same error.

**Comment by nijel** [comment]
You have a few options:

* Stop using SSL/LTS for e-mail
* Use a certificate signed by a trusted CA
* Use EMAIL_SSL_CERTFILE to configure server certificate in Weblate

**Reply by uSkizzik** [reply]
I assume the certificate is setup in my email service? (modoboa)

**Reply by uSkizzik** [reply]
Also how would I use EMAIL_SSL_CERTFILE? What value do I pass? (I assume it would be a path to the file?) Where do I pass it to?

**Reply by nijel** [reply]
Path to file, see https://docs.djangoproject.com/en/4.2/ref/settings/#email-ssl-certfile. In case you are using Docker, place the file in the data volume and then reference it as `/app/data/...`.

**Reply by uSkizzik** [reply]
Sorry, never had to manually interact with volumes. How do I put it in the volume?

**Reply by uSkizzik** [reply]
~~I just realized that's kinda not possible because I'm using let's encrypt and as far as I know the let's encrypt certificates change every once in a while, meaning I'll likely have to keep copy pasting the certs...~~

EDIT: And I also realized the certs are not issued by me. I'm using Modoboa and I never really did any SSL setup besides the one on my reverse proxy which affects only the web portal.
Disabling TLS doesn't work either, Modoboa returns "The SMTP server does not support authentication."

**Reply by nijel** [reply]
Let's encrypt certificates are not self-signed, so the issue would not appear with them.

The volume is mounted somewhere from your host file system, you can use `docker volume ls` and `docker volume inspect VOLUME` to figure that location out. See https://docs.docker.com/storage/volumes/

**Reply by uSkizzik** [reply]
I'll likely have to get in touch with the guys over at the modoboa project to figure out how I can use let's encrypt certs with that.
Thanks for the help.
